src/components/roar-page/useHeader.tsx

```diff
@@ -1,8 +1,25 @@
+/*
```

Comment:
Most of the time one big comment at the beginning is enough, but the code is a bit harder to read and I believe a few inline comments would make a big difference.